code/game/objects/machinery/nuclearbomb.dm

```diff
@@ -205,125 +213,134 @@
 		else
 			status = "Auth. S1-[safe_text]"
 
-	var/html = {"
-	<b>Nuclear Fission Explosive</b><br />
-	<hr />
-	<b>Status</b>: [status]<br />
-	<b>Timer</b>: [timeleft]<br />
-	<br />
-	Timer: [timer_enabled ? "On" : "Off"] [has_auth ? "<a href='?src=[REF(src)];timer=1'>Toggle</a>" : "Toggle"] <br />
-	Time: [has_auth ? "<a href='?src=[REF(src)];time=-10'>--</a> <a href='?src=[REF(src)];time=-1'>-</a> [timeleft] <a href='?src=[REF(src)];time=1'>+</a> <a href='?src=[REF(src)];time=10'>++</a>" : "- - [timeleft] + +"] <br />
-	<br />
-	Safety: [safety ? "On" : "Off"] [has_auth ? "<a href='?src=[REF(src)];safety=1'>Toggle</a>" : "Toggle"] <br />
-	Anchor: [anchored ? "Engaged" : "Off"] [has_auth ? "<a href='?src=[REF(src)];anchor=1'>Toggle</a>" : "Toggle"] <br />
-	<hr />
-	Red Auth. Disk: <a href='?src=[REF(src)];disk=red'>[r_auth ? "++++++++++" : "----------"]</a><br />
-	Green Auth. Disk: <a href='?src=[REF(src)];disk=green'>[g_auth ? "++++++++++" : "----------"]</a><br />
-	Blue Auth. Disk: <a href='?src=[REF(src)];disk=blue'>[b_auth ? "++++++++++" : "----------"]</a><br />
-	"}
-
-	var/datum/browser/popup = new(user, "nuclearbomb", "<div align='center'>Nuclear Bomb</div>", 300, 400)
-	popup.set_content(html)
-	popup.open()
-
-
-
-/obj/machinery/nuclearbomb/Topic(href, href_list)
+	if(timeleft <= 5)
+		status = "Initializing nuclear explosion. Have a nice day :)"
+
+	data["status"] = status
+
+	return data
+
+/obj/machinery/nuclearbomb/ui_act(action, list/params)
 	. = ..()
 	if(.)
 		return
+	switch(action)
+		if("toggle_timer")
+			toggle_timer()
+		if("change_time")
+			change_time(params["seconds"])
+		if("toggle_safety")
+			toggle_safety()
+		if("toggle_anchor")
+			toggle_anchor()
+		if("toggle_disk")
+			toggle_disk(params["disktype"])
```

Comment:
hasauth should be checked here and not in the individual procs for toggling

Reply:
done